kpops/components/base_components/kafka_connector.py

```diff
@@ -94,6 +94,18 @@ class KafkaConnector(PipelineComponent, ABC):
     class Config(CamelCaseConfig):
         pass
 
+    @validator("app", pre=True)
+    def connector_config_should_have_component_name(
+        cls, v: KafkaConnectorConfig | dict[str, str], values: dict[str, Any]
+    ) -> dict[str, str]:
+        if isinstance(v, KafkaConnectorConfig):
+            v = v.dict()
+        connector_name: str | None = v.get("name")
+        if connector_name and connector_name != values["name"]:
```

Comment:
wouldn't be better to use `if connector_name is not None` ?

Reply:
What if the value is `""`?

Reply:
I kinda prefer checking if the str is not empty as well, as we know by convention that every component has to have a name – so I don't think it matters

Reply:
oh no, you're right

Reply:
thanks! I've changed it and added a test